sympy/physics/quantum/grover.py

```diff
@@ -324,3 +324,112 @@ def apply_grover(oracle, nqubits, iterations=None):
         iterated = qapply(iterated)
 
     return iterated
+
+
+def random_oracle(nqubits, min_img=1, max_img=1, q_type='bin'):
+    """Create a random OracleGate under the given parameter
+
+    Parameters
+    ==========
+
+    nqubits : int
+        The number of qubits for OracleGate
+    min_pic : int
+        Minimum number of invers images that are mapped to 1
+    max_pic : int
+        Maximum number of invers images that are mapped to 1
+    q_type : OracleGate
+        Type of the Qubits that the oracle should be applied on.
+        Can be 'bin' for binary (Qubit()) or 'int' for integer (IntQubit()).
+
+    Returns
+    =======
+
+    OracleGate : random OracleGate under the given parameter
+
+    Examples
+    ========
+
+    Generate random OracleGate that outputs 1 for 2-4 inputs::
+
+        >>> from sympy.physics.quantum.grover import random_oracle
+        >>> oracle = random_oracle(4, min_img=2, max_img=4, q_type="bin")
+
+    """
+    if q_type != 'bin' and q_type != 'int':
+        raise QuantumError("q_type must be 'int' or 'bin'")
+
+    if min_img < 1 or max_img < 1:
+        raise QuantumError("min_pic, max_pic must be > 0")
+
+    if min_img > max_img:
+        raise QuantumError("max_pic must be >= min_pic")
+
+    if min_img >= 2 ** nqubits or max_img > 2 ** nqubits:
+        raise QuantumError("min_pic must be < 2**nqubits and max_pic must be <= 2**nqubits")
+
+    import random
+
+    integers = list(range(nqubits ** 2))
```

Comment:
Should this have `2**nqubits`? If so, then the list may take a lot of space. It would be more efficient to use something like `random.sample(range(2**nqubits), pics)` to create a list of random integers.

Reply:
Thanks for the tip. On the danger that I now seem stupid: why is it so much faster? (I tried it, it's about 7 times faster)